### PR TITLE
Staging fixing12

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:

CL: [Improvement] You can press one of the chat modifiers when pressing M to turn off whisper to have it post automatically to the channel. works for (;,b,u,c,e,v)
CL: [Improvement] when chat window is open you can press F6 to toggle whisper
CL: [Fix] fixed light replacer logic, Light replace it could break if you Insert a Lightbulb and then Light tubes and then try to replace light tubes
CL: [Fix] tmpe workaround for Linux crash with whisper, whispers only initialised when pressing m for the first timee

### CODE REVIEW

File: UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs [hunk 1]
```diff
@@ -57,6 +65,20 @@ private async void OnRecordStop(AudioChunk recordedAudio)
...
 
+		var ToUesChatChannel = ChatChannel.Local;
+		if (InputManagerWrapper.GetKey(KeyCode.Semicolon)) ToUesChatChannel |= ChatChannel.Common;
+		if (InputManagerWrapper.GetKey(KeyCode.B)) ToUesChatChannel |= ChatChannel.Binary;
+		if (InputManagerWrapper.GetKey(KeyCode.U)) ToUesChatChannel |= ChatChannel.Supply;
+		//if (InputManagerWrapper.GetKey(KeyCode.Y)) ToUesChatChannel |= ChatChannel.CentComm; Conflicts with opening chat with Local Preselected
+		if (InputManagerWrapper.GetKey(KeyCode.C)) ToUesChatChannel |= ChatChannel.Command;
+		if (InputManagerWrapper.GetKey(KeyCode.E)) ToUesChatChannel |= ChatChannel.Engineering;
+		//if (InputManagerWrapper.GetKey(KeyCode.M)) ToUesChatChannel |= ChatChannel.Medical; //Conflicts with toggling STT (This very thing )
+		//if (InputManagerWrapper.GetKey(KeyCode.N)) ToUesChatChannel |= ChatChannel.Science; //Conflicts with toggle voice chat
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs:76]: bafkreigtavdm2yj4wqlq7e42jcuixlz4xwt3ugvxxbcqrwnxangqt7xsbe

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs:76]: Comments

Comment (Bod9001) [reply UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs:76]: well explains why x channel is not Implemented

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs [hunk 2]
```diff
@@ -620,7 +620,7 @@ private void ExternalMetaboliseReactions()
 				{
 					foreach (var surfaceReagent in SurfaceReagents)
 					{
-						ApplyReagentsToSurface(node.SmokeNode.Present.Clone(), surfaceReagent.Key);
+						ApplyReagentsToSurface(node.FoamNode.Present.Clone(), surfaceReagent.Key);
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs:623]: why would mobs have foam on their surfaces? also what happens if we have different surface reagents that isn't foam?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs:623]: well if foam is capable of making walls I imagine it's like fills up the entire Tile and you have to walk through it

File: UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs:76]: this is a terrible hack/workaround

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs:623]: surface reagents ? what do you mean like chemical patch?

File: UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs:76]: Problem is the chat chapbox shortcuts are Hard coded too

Comment (Bod9001) [reply UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs:76]:
```
private static ChatChannel CheckRadio(string playerInput, IChatInputContext context, out int specialCharCount)

if (firstLetter == ';')
	
	ChannelsTags[secondLetter];

public static readonly Dictionary<char, ChatChannel> ChannelsTags = new Dictionary<char, ChatChannel>
{
   {'b', ChatChannel.Binary},
{'u', ChatChannel.Supply},
```

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/whisper/WhisperMicrophoneHandler.cs:76]: GM-tXBnaYAAeBGE

**Review by MaxIsJoe — APPROVED** [review]

**Pull request merged**